sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -302,7 +302,12 @@ class CustomerIO internal constructor(
      * Use to provide additional and custom device attributes
      * apart from the ones the SDK is programmed to send to customer workspace.
      */
-    val deviceAttributes: MutableMap<String, Any> = mutableMapOf()
+    var deviceAttributes: CustomAttributes = emptyMap()
```

Comment:
>This PR changes the API to syntax that is more familiar to Kotlin developers

This isn't just a syntax change, this changes the behavior of how `CustomAttributes` will be set. 

I intentionally kept it as a `MutableMap`, because I didn't want users to be restricted to provide all the information at once or come up with a mechanism to store all information in one place and then send it to us. The proposed change removes that capability. 

```
// On Sign-up page 1,
CustomerIO.instance().deviceAttributes.putAll(mapOf("language" to "english"))
// On Sign-up page 2           
CustomerIO.instance().deviceAttributes.putAll(mapOf("ads" to false))
           
// Now when you send a device token all of these stored attributes can go along with it.    

Whereas,  in the proposed change, user will have to do it in one shot
CustomerIO.instance().deviceAttributes = mapOf()      
```

Reply:
The use case mentioned in the comment above is still covered by this PR. 

```
// On Sign-up page 1,
CustomerIO.instance().deviceAttributes = mapOf("language" to "english")
// Device attributes "language" get registered to the device on the API

// On Sign-up page 2           
CustomerIO.instance().deviceAttributes = mapOf("ads" to false)
// Device attributes "ads" get registered to the device on the API
```

I don't see a problem with the above use case? Customers still have the convenience of not having to add all of the device attributes in one shot. Plus they get the added benefits outlined in the description of this PR in the "Why this change is being proposed" section.

Reply:
You are correct. The use case of custom device attributes being ignored in the SDK when a device token has not been registered [has been documented and the squad has had conversations together about](https://github.com/customerio/issues/issues/6982). 

We as a squad decided that we want to ignore custom device attributes when a push token is not yet registered. We will make a note of this in the SDK docs. Then, we will collect customer feedback and make changes to the SDK behavior in the future if we choose to do so.

Reply:
Alright, I think I missed this conversation as I must not be working that day. But by just making `deviceAttributes` a `MutableMap` and then emptying it when the task is added to BQ can easily solve the mentioned use-case. 

>The use case of custom device attributes being ignored in the SDK when a device token has not been registered

I'll unblock this PR but I do believe we should move to solve this as it's a very quick fix and doesn't `Why this change is being proposed` part as well.